OpenRA.Mods.Common/Traits/Render/RenderSprites.cs

```diff
@@ -236,19 +237,38 @@ public static string NormalizeSequence(Animation anim, DamageState state, string
 			return sequence;
 		}
 
-		// Required by WithSpriteBody and WithInfantryBody
 		public int2 AutoSelectionSize(Actor self)
 		{
-			return AutoRenderSize(self);
+			var visibleAnims = anims.Where(b => b.IsVisible && b.Animation.Animation.CurrentSequence != null).ToArray();
+			var aggregateSize = int2.Zero;
+			foreach (var a in visibleAnims)
+			{
+				var size = (a.Animation.Animation.Image.Size.XY * info.Scale).ToInt2();
+				if (size.X * size.X > aggregateSize.X * aggregateSize.X)
+					aggregateSize = new int2(size.X, aggregateSize.Y);
+				if (size.Y * size.Y > aggregateSize.Y * aggregateSize.Y)
+					aggregateSize = new int2(aggregateSize.X, size.Y);
+			}
+
+			return aggregateSize;
 		}
 
-		// Required by WithSpriteBody and WithInfantryBody
-		public int2 AutoRenderSize(Actor self)
+		Rectangle IRender.AutoRenderBounds(Actor self)
 		{
-			return anims.Where(b => b.IsVisible
-				&& b.Animation.Animation.CurrentSequence != null)
-					.Select(a => (a.Animation.Animation.Image.Size.XY * info.Scale).ToInt2())
-					.FirstOrDefault();
+			var visibleAnims = anims.Where(b => b.IsVisible && b.Animation.Animation.CurrentSequence != null).ToArray();
```

Comment:
This is tricky, because changing the visible animations or the sequence of currently visible animations can change the screen bounds of the actor.  We will need to adopt @penev92's idea for notifying/triggering a bounds recalculation when the visible animations change.